sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj

```diff
@@ -50,6 +50,7 @@
     <Compile Include="$(TestFrameworkSupportFiles)" LinkBase="Shared\TestFramework" />
     <Compile Include="..\src\Shared\ApiVersionString.cs" LinkBase="Shared" />
     <Compile Include="..\src\Shared\GuidUtilities.cs" LinkBase="Shared" />
+    <Compile Remove="public/**" />
```

Comment:
This is odd.   What's `public` and what are we up to?

Reply:
The public directory holds a test project that doesn't have `InternalsVisibleTo` granted from Azure.Core.  That allows us to test the dynamic type like an end-user, which we can't do in the standard Azure.Core.Tests project.  Since some tests fail by design (e.g. testing that a user of the public API doesn't have access to internal methods), we can't include those tests in this project.

Reply:
Thanks, that makes sense.

It does feel a bit odd to me that the hierarchy represents them as a child of the test project rather than sibling.   Nothing wrong as-is, but it may be worth considering moving the "public" project file up to be a sibling to the main test project and letting "Public" just be another set of tests in the hierarchy (albeit ones that aren't referenced by the main test project)

Reply:
That's fair.  I looked at this part early on in the JsonData project, and I think this was the only way I could get what I wanted to do working, but I've learned a bit about the test infra since then, so let me take another pass on it.  If I can't do what your suggesting for some reason, hopefully I can do better at explaining why :)

Reply:
Hm, unfortunately there are a lot of tests in Azure.Core.Tests that rely on `InternalVisibleTo`, so moving `Azure.Core.Tests.Public.csproj` up a directory results in a lot of CS0122 build errors.  I'm not sure I see the value of sifting through these and partitioning existing Azure.Core tests into internal/public, or even if that was the goal of your comment, @jsquire?

Can we pop up a level and articulate what we'd be trying to accomplish with this move, @jsquire?  I'm totally game to improve the situation, but want to understand the goal state better.

Reply:
My goal was just to try and organize the tests and public tests as siblings in the hierarchy rather than have `public` hide another project, which makes it look like a child.   To do that, you'd have to do a remove all for other types and just include the `public` subdirectory, while still excluding `public` from the standard tests.

```text
$/sdk/core/Azure.Core/tests
    | -- Azure.Core.Tests.csproj
    | -- /Buffers 
            | -- SomeTest.cs  (belongs to Azure.Core.Tests.csproj)
    | -- /samples 
            | -- SomeTests.cs  (belongs to Azure.Core.Tests.csproj)
    | -- /public            
            | -- Azure.Core.Public.Tests.csproj  (does NOT belong to Azure.Core.Tests.csproj)
            | -- SomeTest.cs  (does NOT belong to Azure.Core.Tests.csproj)           
```

versus

```text
$/sdk/core/Azure.Core/tests
    | -- Azure.Core.Tests.csproj
    | -- Azure.Core.Public.Tests.csproj
    | -- /Buffers 
            | -- SomeTest.cs  (belongs to Azure.Core.Tests.csproj)
    | -- /samples 
            | -- SomeTests.cs  (belongs to Azure.Core.Tests.csproj)
    | -- /public            
            | -- SomeTest.cs  (belongs to Azure.Core.Public.Tests.csprojj)           
```

To me, the second one of these makes more sense and would help me understand more easily that we have two different projects for tests.   The first makes it harder for me to know why I may not see all the tests that I think I should.    It's true that the view in VS would normalize the structure, but VS Code would not.

Functionally, there's no difference and I wouldn't call it important.

Reply:
Definitely non-blocking.  If there's complexity, I'd agree that the ROI isn't there and we should just drop it.   No need to track via issue.